views/default/input/date.js

```diff
@@ -0,0 +1,62 @@
+define(function (require) {
+
+	var elgg = require('elgg');
+	var $ = require('jquery');
+	require('jquery-ui');
+
+	if (elgg.get_language() !== 'en') {
+		// require language module if necessary
+		require('jquery-ui/i18n/datepicker-' + elgg.get_language() + '.min');
```

Comment:
This can't work. 1. It would need to be `require([' ... '])` to load async. 2. We have to guarantee the language module is loaded before `$elem.datepicker()` is called.

I think I have a solution I can post tomorrow. Basically we'll use promises to delay the `$elem.datepicker()` call until the language is loaded in an async `require` call.